File: Persistencia/Scripts/micro_dao/info_profissional.py
from datetime import datetime

from micro_dao.objeto_modelo import ObjetoModelo

class InfoProfissional(ObjetoModelo):


    _id: int
    _descricao: str
    _data_inicio: datetime
    _data_fim: datetime
    _fk_usuario: int
    _fk_tipo_info_profissional: int

    def __init__(self, descricao: str, data_inicio: datetime, data_fim: datetime,
                 fk_usuario: int, fk_tipo_info_pro: int):
        self.set_descricao(descricao)
        self.set_data_inicio(data_inicio)
        self.set_data_fim(data_fim)
        self.set_fk_usuario(fk_usuario)
        self.set_fk_tipo_info_prof(fk_tipo_info_pro)

    def get_id(self) -> int:
        return self._id

    def set_id(self, id: int):
        self._id = id
        return self

    def get_descricao(self) -> str:
        return self._descricao

    def set_descricao(self, descricao: str):
        self._descricao = descricao
        return self

    def get_data_inicio(self) -> datetime:
        return self._data_inicio

    def set_data_inicio(self, data_inicio: datetime):
        self._data_inicio = data_inicio
        return self

    def get_data_fim(self) -> datetime:
        return self._data_fim

    def set_data_fim(self, data_fim: datetime):
        self._data_fim = data_fim
        return self

    def get_fk_usuario(self) -> int:
        return self._fk_endereco

    def set_fk_usuario(self, fk_endereco: int):
        self._fk_endereco = fk_endereco
        return self

    def get_fk_tipo_info_prof(self) -> int:
        return self._fk_tipo_info_profissional

    def set_fk_tipo_info_prof(self, fk_tipo_info_pro: int):
        self._fk_tipo_info_profissional = fk_tipo_info_pro
        return self

File: Projeto_Java/src/Dominio/Entidades/InfoProfissional.java
package Dominio.Entidades;

import Dominio.Enum.ETipoInfoProfissional;
import Dominio.Util.Util;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "info_profissional")
public class InfoProfissional {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "descricao", length = 200)
    private String descricao;

    @Column(name = "data_inicio")
    private LocalDate dataInicio;

    @Column(name = "data_fim")
    private LocalDate dataFim;

    @Transient
    private ETipoInfoProfissional tipoInfo;

    @Column(name = "fk_tipo_info_prof")
    private int fkTipoInfo;

    @Column(name = "fk_usuario")
    private int fkUsuario;


    public InfoProfissional() { }

    public InfoProfissional(String descricao, LocalDate dataInicio, LocalDate dataFim,
                            int fkTipoInfo, int fkUsuario) {
        this.setDescricao(descricao);
        this.setDataInicio(dataInicio);
        this.setDataFim(dataFim);
        this.setFkTipoInfo(fkTipoInfo);
        this.setFkUsuario(fkUsuario);
    }

    public InfoProfissional(int id, String descricao, LocalDate dataInicio, LocalDate dataFim,
                            int fkTipoInfo, int fkUsuario) {
        this.setId(id);
        this.setDescricao(descricao);
        this.setDataInicio(dataInicio);
        this.setDataFim(dataFim);
        this.setFkTipoInfo(fkTipoInfo);
        this.setFkUsuario(fkUsuario);
    }


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public ETipoInfoProfissional getTipoInfoProfissional() {
        return tipoInfo;
    }

    public void setTipoInfo(ETipoInfoProfissional tipoInfo) {
        this.tipoInfo = tipoInfo;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {

        if (Util.isStringValida(descricao)) this.descricao = descricao;

        else Util.throwExceptCampoVazio("descrição");
    }

    public LocalDate getDataInicio() {
        return dataInicio;
    }

    public void setDataInicio(LocalDate dataInicio) {
        this.dataInicio = dataInicio;
    }

    public LocalDate getDataFim() {
        return dataFim;
    }

    public void setDataFim(LocalDate dataFim) {
        this.dataFim = dataFim;
    }

    public int getFkUsuario() {
        return fkUsuario;
    }

    private void setFkUsuario(int fkUsuario) {
        this.fkUsuario = fkUsuario;
    }

    public int getFkTipoInfo() {
        return fkTipoInfo;
    }

    public void setFkTipoInfo(int fkTipoInfo) {
        setTipoInfo(ETipoInfoProfissional.getById(fkTipoInfo));
        this.fkTipoInfo = fkTipoInfo;
    }

    @Override
    public String toString() {

        String enderecoStr = "\nID:\t\t\t\t" + getId();
        enderecoStr += "\nDescrição:\t\t" + getDescricao();
        enderecoStr += "\nData início:\t" + getDataInicio();
        enderecoStr += "\nData fim:\t\t" + getDataFim();
        enderecoStr += "\nTipo info.:\t\t" + getTipoInfoProfissional().getTipo();
        enderecoStr += "\nFK_tipo_info:\t" + getFkTipoInfo();
        enderecoStr += "\nFK_usuario:\t\t" + getFkUsuario();

        return enderecoStr;
    }
}
